src/commands/registerCommands.ts

```diff
@@ -87,25 +88,83 @@ async function registerCodeLensCommands(context: ExtensionContext, languageClien
       then(() => {
         window.showInformationMessage('The current XML file was successfully validated.');
       }, error => {
         window.showErrorMessage('Error during XML validation ' + error.message);
       });
   }));
   // Revalidate all open files
-  context.subscriptions.push(commands.registerCommand(CommandConstants.VALIDATE_ALL_FILES, async () => {
-    commands.executeCommand(CommandConstants.EXECUTE_WORKSPACE_COMMAND, CommandConstants.VALIDATE_ALL_FILES).
+  context.subscriptions.push(commands.registerCommand(ClientCommandConstants.VALIDATE_ALL_FILES, async () => {
+    commands.executeCommand(ClientCommandConstants.EXECUTE_WORKSPACE_COMMAND, ServerCommandConstants.VALIDATE_ALL_FILES).
       then(() => {
         window.showInformationMessage('All open XML files were successfully validated.');
       }, error => {
         window.showErrorMessage('Error during XML validation: ' + error.message);
       });
   }));
+}
+
+export const bindingTypes = new Map<string, string>([
+  ["Standard (xsi, DOCTYPE)", "standard"],
+  ["XML Model association", "xml-model"]
+]);
+
+const bindingTypeOptions: QuickPickItem[] = [];
+for (const label of bindingTypes.keys()) {
+  bindingTypeOptions.push({ "label": label });
+}
+
+/**
+ * Register commands used for associating grammar file (XSD,DTD) to a given XML file
+ *
+ * @param context the extension context
+ */
+function registerCodeLensAssociationCommands(context: ExtensionContext, languageClient: LanguageClient) {
+  context.subscriptions.push(commands.registerCommand(ClientCommandConstants.OPEN_BINDING_WIZARD, async (uriString: string) => {
+    // A click on Bind to grammar/schema... has been processed in the XML document which is not bound to a grammar
+    const documentURI = Uri.parse(uriString);
+
+    // Step 1 : open a combo to select the binding type ("standard", "xml-model")
+    const pickedBindingTypeOption = await window.showQuickPick(bindingTypeOptions, { placeHolder: "Binding type" });
+    if(!pickedBindingTypeOption) {
+      return;
+    }
+    const bindingType = bindingTypes.get(pickedBindingTypeOption.label);
+
+    // Open a dialog to select the XSD, DTD to bind.
+    const options: OpenDialogOptions = {
+      canSelectMany: false,
+      openLabel: 'Select XSD or DTD file',
+      filters: {
+        'Grammar files': ['xsd', 'dtd']
+      }
+    };
+
+    const fileUri = await window.showOpenDialog(options);
+    if (fileUri && fileUri[0]) {
+      // The XSD, DTD has been selected, get the proper syntax for binding this grammar file in the XML document.
+      const identifier = TextDocumentIdentifier.create(documentURI.toString());
+      const grammarURI = fileUri[0];
+      try {
+        const result = await commands.executeCommand(ClientCommandConstants.EXECUTE_WORKSPACE_COMMAND, ServerCommandConstants.ASSOCIATE_GRAMMAR_INSERT, identifier, grammarURI.toString(), bindingType);
```

Comment:
I still don't think you need `EXECUTE_WORKSPACE_COMMAND`, it works fine if you directly call the `ASSOCIATE_GRAMMAR_INSERT` command.

Reply:
Let me experiment that i dont understand how it could work?

Reply:
Any chance that you could address this?

Reply:
Indeed it works, I updated it. It works because LemMinX returns the ServerCommandConstants.ASSOCIATE_GRAMMAR_INSERT (xml.associate.grammar.insert) as workspace command:

```
[Trace - 17:10:40] Received request 'client/registerCapability - (19)'.
Params: {
    "registrations": [
        {
            "id": "157b25db-dfea-4a42-bdea-953be34d71b3",
            "method": "workspace/executeCommand",
            "registerOptions": {
                "commands": [
                    "xml.associate.grammar.insert",
                    "xml.validation.current.file",
                    "xml.validation.all.files"
                ]
            }
        }
    ]
}

```

I tried to the same thing for `xml.validation.current.file` but it doesn't work. I suppose it's because we use the same command ID in the package.json to register the command than xml.validation.current.file.

Could you create an issue about that please.